internal/service/advancedclustertpf/model_to_ClusterDescriptionProcessArgsLegacy.go

```diff
@@ -5,24 +5,25 @@ import (
 
 	"github.com/hashicorp/terraform-plugin-framework/diag"
 	"github.com/hashicorp/terraform-plugin-framework/types"
-	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
 	"github.com/mongodb/terraform-provider-mongodbatlas/internal/common/conversion"
 	admin20240530 "go.mongodb.org/atlas-sdk/v20240530005/admin"
 )
 
-func NewAtlasReqAdvancedConfigurationLegacy(ctx context.Context, objInput *types.Object, diags *diag.Diagnostics) *admin20240530.ClusterDescriptionProcessArgs {
+func NewAtlasReqAdvancedConfigurationLegacy(ctx context.Context, input *types.List, diags *diag.Diagnostics) *admin20240530.ClusterDescriptionProcessArgs {
 	var resp *admin20240530.ClusterDescriptionProcessArgs
-	if objInput == nil || objInput.IsUnknown() || objInput.IsNull() {
+	if input == nil || input.IsUnknown() || input.IsNull() || len(input.Elements()) == 0 {
```

Comment:
tempting to create a helper: `listHasValues(input *types.List) bool`

Reply:
@EspenAlbert WDYT about this refactor? https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2909/commits/a829bf64112a58ec8e429d40d9115f1a47822648

Reply:
Love it!